scripts/configs.js

```diff
@@ -1,85 +1,131 @@
+}
 
-  const environmentFromConfig = window.sessionStorage.getItem('environment');
-  if (
-    environmentFromConfig
-    && ALLOWED_CONFIGS.includes(environmentFromConfig)
-    && environment !== 'prod'
-  ) {
-    return environmentFromConfig;
+/**
+ * Retrieves a value from a config object using dot notation.
+ *
+ * @param {Object} obj - The config object.
+ * @param {string} key - The key to retrieve (supports dot notation).
+ * @returns {any} - The value of the key.
+ */
+function getValue(obj, key) {
+  return key.split('.').reduce((current, part) => {
+    if (!Object.prototype.hasOwnProperty.call(current, part)) {
+      console.warn(`Property ${key} does not exist in the object`);
+      return undefined;
+    }
+    return current[part];
+  }, obj);
+}
+
+/**
+ * Get root path
+ * @param {Object} [config] - The config object.
+ * @returns {string} - The root path.
+ */
+export function getRootPath() {
+  const value = Object.keys(CONFIG?.public)
+    // Sort by number of non-empty segments to find the deepest path
+    .sort((a, b) => {
+      const aSegments = a.split('/').filter(Boolean).length;
+      const bSegments = b.split('/').filter(Boolean).length;
+      return bSegments - aSegments;
+    })
+    .find((key) => window.location.pathname === key || window.location.pathname.startsWith(key));
+
+  const rootPath = value ?? '/';
+
+  if (!rootPath.startsWith('/') || !rootPath.endsWith('/')) {
+    throw new Error('Invalid root path');
   }
 
-  return environment;
-};
+  return rootPath;
+}
 
-function buildConfigURL(environment) {
-  const env = environment || calcEnvironment();
-  let fileName = 'configs.json';
-  if (env !== 'prod') {
-    fileName = `configs-${env}.json`;
-  }
-  const configURL = new URL(`${window.location.origin}/${fileName}`);
-  return configURL;
+/**
+ * Get list of root paths from public config
+ * @returns {Array} - The list of root paths.
+ */
+export function getListOfRootPaths() {
+  return Object.keys(CONFIG?.public).filter((root) => root !== 'default');
 }
 
-function applyConfigOverrides(config) {
-  // get overrides
-  const website = getMetadata('commerce-website');
-  const store = getMetadata('commerce-store');
-  const storeview = getMetadata('commerce-storeview');
-
-  // add overrides
-  const updates = new Map([
-    ['commerce.headers.cs.Magento-Website-Code', website],
-    ['commerce.headers.cs.Magento-Store-Code', store],
-    ['commerce.headers.cs.Magento-Store-View-Code', storeview],
-    ['commerce.headers.all.Store', storeview],
-  ]);
-
-  // apply updates
-  config.data.forEach((item) => {
-    const next = updates.get(item.key);
-    if (next) {
-      item.value = next;
-      updates.delete(item.key);
-    }
-  });
+/**
+ * Checks if the public config contains more than "default"
+ * @returns true if public config contains more than "default"
+ */
+export function isMultistore() {
+  return getListOfRootPaths().length > 1;
```

Comment:
`getListOfRootPaths` already returns the list without default. So I think this needs to be `>= 1`?

```suggestion
  return getListOfRootPaths().length >= 1;
```